src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +38,43 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
+        checkUniqueInstance(params); //java 9
         checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
+    private static void checkUniqueInstance(StartupParams params) throws IOException {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+
+        if (Files.exists(filePath)) {
+            List<String> fileContent = new ArrayList<>(Files.readAllLines(filePath));
+
+            if (fileContent.isEmpty()) {
+                Files.writeString(filePath, "\n" + currentPid, StandardOpenOption.APPEND);
+            } else {
+                for (int i = 0; i < fileContent.size(); i++) {
+                    if (ProcessHandle.of(Long.parseLong(fileContent.get(i))).isPresent()) {
+                        Files.writeString(filePath, "\n" + currentPid, StandardOpenOption.APPEND);
```

Comment:
does this mean we only write *if* there's a collision?
or otherwise, have a completely separate if statement to handle the same thing (writing to file) ?

Just do this after the whole checking section, and only if the user decides to proceed (if they shutdown instead, no need). Also don't append, simply overwrite if it exists already, Create new if missing. i believe Files.writeString already has an option for it. It's simpler and we don't need to keep track of all processes that are created as long as we've warned them at least once.